verify_mapping_quality.pl

```diff
@@ -0,0 +1,229 @@
+#!/usr/bin/env perl
+use strict;
+use warnings;
+use Pod::Usage;
+use Getopt::Long;
+use autodie;
+use FindBin qw($Bin);
+use File::Basename;
+use Parallel::ForkManager;
+use Readonly;
+
+Readonly my $MIN_DEPTH => 10;    # THIRD COLUMN
+Readonly my $MIN_MAP => 80;
+
+__PACKAGE__->run unless caller;
+
+1;
+
+sub run {
+    my ( $size, $man, $help, $min_depth, $log_dir, $min_map, %bam_files, $cores );
+	
+    GetOptions(
+        "c|cores=i"   => \$cores,
+        "bam=s"		  => \%bam_files,
+        "l|log_dir=s" => \$log_dir,
+        "s|size=s"    => \$size,
+        "min-map=f"	  => \$min_map,
+        "min-depth=i" => \$min_depth,
+        "h|help"      => \$help,
+        "m|man"       => \$man
+    );
+    pod2usage(1) if $help;
+    pod2usage(-verbose => 2) if $man;
+	
+    unless ( (scalar keys %bam_files != 0 ) ) {
+        print "Unable to find any input bam files.\n\n";
+        pod2usage(1);
+    }
+    unless ( defined $size ) {
+        print "Calculating genome size from input BAM files.\n\n";
+    }
+		
+	#set default values if undefined on command line
+    if(!defined $min_depth){
+        $min_depth = $MIN_DEPTH;
+        print "  Using a minumum depth of $MIN_DEPTH\n";
+        print "  Use the --min-depth flag to set custom depth\n\n";
+    }
+    else{
+    	print "Using a min_depth of ".$min_depth."\n";
+    }
+    #set default number of cores
+	$cores = 1 if (not defined $cores);
+	#set default log directory to local directory    
+	$log_dir="./" if (not defined $log_dir);
+	#set default minimum percent mapping
+	$min_map=$MIN_MAP if (not defined $min_map);
+	
+	#create the log file to print warnings to
+	open(my $log, '>', $log_dir.'mapping_percentage.log');
+	
+	#retrieve all of the bam file locations from the hash
+    my @files = values %bam_files;
+ 
+ 	#ensure that bam files are properly input on the command line and that each file path exists
+ 	if (@files <= 0){ die "Error: No bam files input."};
+ 	foreach(@files){
+ 		if (!-e $_) {die "Error: Invalid bam file referenced."};
+ 	}
+	#command to get the size of the genome from the bam file:
+	my $command = `samtools view -H $files[0] | grep -P '^\@SQ' | cut -f 3 -d ':' | awk '{sum+=$1} END {print sum}'`;
+	$size = system($command);
+	#ensure the size of the genome is properly determined
```

Comment:
You are calling your samtools command twice. First time is when you use the `` quotes on line 71. Reason you are getting the Use of uninitialized value $1 is that Perl thinks you are asking for first group match from a regular expression. Fixed that by escaping $1 i.e '\$1' . 

Then you are taking the result from `` and then calling system. 

One way to fix would be to get rid of line 72, escape $1 and change my $command to $size on line 71.